src/test/scala/tech/cryptonomic/conseil/tezos/TezosRemoteInstancesTest.scala

```diff
@@ -0,0 +1,107 @@
+package tech.cryptonomic.conseil.tezos
+
+import org.scalatest.{BeforeAndAfterAll, Matchers, WordSpecLike}
+import scala.concurrent.duration._
+import akka.actor.ActorSystem
+import akka.testkit.TestKit
+import akka.stream.ActorMaterializer
+import tech.cryptonomic.conseil.generic.rpc.RpcHandler
+import cats.effect.IO
+import com.typesafe.config.ConfigFactory
+import tech.cryptonomic.conseil.util.JsonUtil
+
+class TezosRemoteInstancesTest
+    extends TestKit(ActorSystem("LocalWebServer"))
+    with WordSpecLike
+    with Matchers
+    with BeforeAndAfterAll {
+
+  override def afterAll(): Unit = {
+    TestKit.shutdownActorSystem(system)
+    super.afterAll()
+  }
+
+  "The IOEff RpcHander" should {
+      import TezosRemoteInstances.Cats.IOEff._
+
+      "GET json data from a web server" in withServerReturning(dummyJsonObject) {
+        //given
+        import LocalNodeContext._
+
+        //when
+        val result: String = RpcHandler.runGet("anypath").unsafeRunSync()
+
+        //then
+        result shouldEqual dummyJsonObject
+      }
+
+      "POST json data from a web server" in withServerReturning(dummyJsonObject) {
+        //given
+        import LocalNodeContext._
+
+        val payload = JsonUtil.JsonString.wrapString("""{"number": 10}""").toOption.ensuring(_.nonEmpty)
+
+        //when
+        val result: String = RpcHandler.runPost("anypath", payload).unsafeRunSync()
+
+        //then
+        result shouldEqual dummyJsonObject
+      }
+    }
+
+  private val dummyJsonObject = """{"name":"dummy"}"""
+
+  /* Provides the contact point info to the test node
+   * Import LocalNodeContext._ to have a TezosNodeContext implicitly in scope
+   */
+  private object LocalNodeContext {
+    import TezosRemoteInstances.Akka.TezosNodeContext
+    import tech.cryptonomic.conseil.config.{HttpStreamingConfiguration, NetworkTimeoutConfiguration, Platforms}
+
+    val tezosConf =
+      Platforms.TezosConfiguration(
+        "local",
+        Platforms.TezosNodeConfiguration("localhost", 9999, "http")
+      )
+
+    val timeouts =
+      NetworkTimeoutConfiguration(GETResponseEntityTimeout = 1.second, POSTResponseEntityTimeout = 1.second)
+
+    val streamingConf = HttpStreamingConfiguration(ConfigFactory.load())
+
+    implicit val nodeContext: TezosNodeContext = TezosNodeContext(tezosConf, timeouts, streamingConf)
+  }
+
+  /* provides a running web server within the test */
+  private def withServerReturning(json: String)(body: => Any) = {
```

Comment:
whny not use wiremock?

Reply:
One less dependency... didn't seem so relevant at the moment, I only needed very basic features.

Might switch in the future if something more flexible is needed.

Reply:
To clarify, I just don't like the way you start and terminate server, beforeEach/afterEach would be better, but I'm not sure if it's possible with your way of testing

Reply:
I find your comment interesting, would you elaborate more, so we can evaluate together pros/cons?

Reply:
What I meant was that if you look at the @dorians implementation of `SttpGithubAuthenticationProviderRepositoryTest` in nautilus-cloud, the server starts once and uses for reseting `beforeEach`.

Reply:
I will try wiremock then, not that crazy a change I think